Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp

```diff
@@ -2120,6 +2121,14 @@ namespace AZ
                     material->GetAsset()->GetMaterialTypeAsset()->GetUvNameMap());
                 AZ_Assert(result, "Failed to retrieve mesh stream buffer views");
 
+                // The code below expects streams for positions, normals, tangents, bitangents, and uvs.
+                constexpr size_t NumExpectedStreams = 5;
+                if (streamBufferViews.size() < NumExpectedStreams)
+                {
+                    AZ_Warning("MeshFeatureProcessor", false, "Model is missing one or more expected streams, skipping.");
```

Comment:
There's _equally_ unhelpful messaging right above this where it asserts and says "Failed to retrieve mesh stream buffer views". ;) I'll change the warning to this: "Model is missing one or more expected streams (positions, normals, tangents, bitangents, uvs), skipping the raytracing data generation." FWIW, I'm not sure how this error would occur in the common use cases though, since the model exporter should produce all of these. I only ran into this when generating my in-memory unit cube model because I only generated positions and normals at first.

Also, I can't speak to why those streams are set to optional above, since I can say from first-hand observation the code used to crash when dereferencing the stream buffer views for those streams if they don't exist. :) I suspect that the code for getting the byte counts and descriptors for the optional streams should probably have guards around them, but at least this code won't crash in the meantime until someone does that (if ever).